RadixWallet/Clients/OnLedgerEntitiesClient/Helpers/OnLedgerEntitiesClient+CreateEntity.swift

```diff
@@ -502,6 +494,34 @@ extension OnLedgerEntity.Account.PoolUnitResources {
 	}
 }
 
+extension [OnLedgerEntity.OwnedNonFungibleResource] {
+	public var filterEmptyVaults: [OnLedgerEntity.OwnedNonFungibleResource] {
+		filter { $0.nonFungibleIdsCount > 0 }
+	}
+}
+
+extension OnLedgerEntity.OwnedFungibleResources {
+	public var filterEmptyVaults: OnLedgerEntity.OwnedFungibleResources {
+		.init(
+			xrdResource: xrdResource.flatMap { $0.amount > 0 ? $0 : nil },
+			nonXrdResources: nonXrdResources.filter { $0.amount > 0 }
+		)
+	}
+}
+
+extension OnLedgerEntity.Account {
+	public var filterEmptyVaults: OnLedgerEntity.Account {
```

Comment:
Now we filter the empty vault resource at reducer level, as with this PR we still need to keep track of resource with empty vaults to know if a given resource was ever owned by the account.